### DISCUSSION TITLE
examples not working

### DISCUSSION THREAD

**Discussion by philhopkins1** [coqui-ai/STT#2289] [General]
I am using Deuvan (the same as debian testing) and I get errors trying to run any of the example from the STT-examples repository. Has anyone tried them recently? I am trying to get a generic webpage running that will change a Sunday sermon to text so the the deaf folks can understand what is going on. I have one working using the google voice recognition api but I would like to try Coqui after I do some training. If anyone will help I will post the errors that I get.

Thanks,
Phil

**Comment by HarikalarKutusu** [comment]
Hey Phil,

Please do post what you are getting inc. you whole setup info... Somebody with similar setup will look it up.

As a side note: Some of the examples are referencing to STT v1.0.0 and might be adjusted as v1.3.0... These do not get corrected everytime a release comes out...

**Reply by philhopkins1** [reply]
OK, the first example that I tried was web_microphone_websocket. I ran yarn start and a browser window opened. Then I ran: node server.js which failed. I got the following error:
Error: Cannot find module '/home/phil/speech/coqui/STT-examples/web_microphone_websocket/node_modules/stt/lib/binding/v1.3.0/linux-x64/node-v108/stt.node' 
With a little more output. When I checked the folder: 
/home/phil/speech/coqui/STT-examples/web_microphone_websocket/node_modules/stt/lib/binding/v1.3.0/linux-x64/
I found a node-v202 but no folder node-v108. I even tried placing a sys link from node-v102 to node-v108 but that failed also with a different error. 

The question I have is why is there no node-v108 folder? How can I fix it?

I will follow this later with the errors that I got when trying python_websocket_server and mic_vad_streaming. I stopped there, it seemed like nothing was working. Let me correct that sst-model-manager does work.

Phil

**Reply by philhopkins1** [reply]
Also i did correct the requirements for stt 1.0.0, I put stt 1.3.0

**Reply by philhopkins1** [reply]
I think that  I see the problem. Whoever built the nodejs stt package needs to build it and upload the latest version for newer systems. The most recent one is for node-v102, whereas for my system I need version node-v108. I cannot install this on my system. I need a simple web server that I can have a browser using javascript to send audio to the server and get the text back, any recommendations?

Phil

**Reply by lissyx** [reply]
`node-v108` is for NodeJS 18.x, there is simply no support for that yet. Use NodeJS 17.x or contribute support for 18.x

**Comment by HarikalarKutusu** [comment]
Interesting, I successfully created a project based on that example on March-April. Just re-cloned the STT-Examples repo and I have these:

![image](https://user-images.githubusercontent.com/8849617/186783017-f2fb7539-9d6d-4c20-91f4-9330e5993e76.png)

And I could run it... I'm on Windows Pro x64 latest thou... And I use npm, node version is old on this notebook, 14.17.6...

It must be related to your setup.

**Reply by HarikalarKutusu** [reply]
Perhaps come to the gitter/matrix channel for rapid interaction?

https://gitter.im/coqui-ai/STT

**Comment by philhopkins1** [comment]
I'll try to switch to NodeJS 17. Thanks.

Phil

**Comment by philhopkins1** [comment]
I tried Ubuntu 20.04 and it failed. If anyone knows of a Linux release that
works please let me know. I can try it using a Virtual machine.

Thanks,
Phil
